state/execution.go

```diff
@@ -425,8 +426,9 @@ func updateState(
 // Fire TxEvent for every tx.
 // NOTE: if Tendermint crashes before commit, some or all of these events may be published again.
 func fireEvents(logger log.Logger, eventBus types.BlockEventPublisher, block *types.Block, abciResponses *ABCIResponses) {
-	eventBus.PublishEventNewBlock(types.EventDataNewBlock{block})
-	eventBus.PublishEventNewBlockHeader(types.EventDataNewBlockHeader{block.Header})
+	resultTags := append(abciResponses.BeginBlock.Tags, abciResponses.EndBlock.Tags...)
+	eventBus.PublishEventNewBlock(types.EventDataNewBlock{block}, resultTags)
```

Comment:
Also, should we make `abciResponses.BeginBlock` and `abciResponses.EndBlock` a part of `EventDataNewBlock` and `EventDataNewBlockHeader`?

Reply:
pros: we won't need a separate event for EventDataValidatorSetUpdates
cons: bigger size of NewBlock & NewBlockHeader structs

Reply:
Yeah I had this implemented previously, but removed it due to your previous comments. So it can be either way, I think having it in `EventDataNewBlock(Header)` is more consistent with `EventDataTx`.

Reply:
Agree. Let's do that. `Also, should we make abciResponses.BeginBlock and abciResponses.EndBlock a part of EventDataNewBlock and EventDataNewBlockHeader`

Reply:
:+1: